iocore/eventsystem/Continuation.cc

```diff
@@ -23,24 +23,22 @@
 
 #include "I_EventSystem.h"
 #include "I_Continuation.h"
-#include "I_EThread.h"
+#include "I_Lock.h"
 
 int
-Continuation::handleEvent(int event, void *data)
-{
-  // If there is a lock, we must be holding it on entry
-  ink_release_assert(!mutex || mutex->thread_holding == this_ethread());
-  return (this->*handler)(event, data);
-}
-
-int
-Continuation::dispatchEvent(int event, void *data)
+Continuation::dispatchEvent(int event, void *data, Event **scheduled_event)
 {
+  if (scheduled_event) {
+    *scheduled_event = nullptr;
+  }
   if (mutex) {
     EThread *t = this_ethread();
     MUTEX_TRY_LOCK(lock, this->mutex, t);
     if (!lock.is_locked()) {
-      t->schedule_imm(this, event, data);
+      Event *sched_event = t->schedule_imm(this, event, data);
+      if (scheduled_event) {
+        *scheduled_event = sched_event;
+      }
```

Comment:
Can we try the following logic to avoid event leaks?
```
if (scheduled_event) {
  *scheduled_event = t->schedule_imm(this, event, data);
} else {
  lock.acquire(t);
  return (this->*handler)(event, data);
}
```

Reply:
Isn't that a blocking lock? That's not allowed in an `ET_NET` thread. If you're going to do a blocking lock at all, it would be better to do that on the original lock. Beyond that, this will call the handler of the continuation without a lock on the continuation's mutex, which is precisely what this PR is trying to prevent. Again, if it's OK to do that, why bother with any of this?

Reply:
Yes, this is a blocking lock.  That would make things much easier if we could block.

Reply:
Doing any sort of blocking lock in the main execution path seems much more fraught that possibly losing events. In the situations I've seen for this, the target continuation is waiting on a reply and therefore should wait until it gets it.

Reply:
@oknet I see your point.  I think I'm going to close this PR and put up separate PRs for putting handleEvent back into the header file (which I think is non-controversial) and another for restructuring dispatchEvent (which probably requires some more discussion).